### PR TITLE
Add Recent Activity page components

### PR DESCRIPTION
**Pull request by pjanik**
This PR adds Recent Activity page components. I'll open related Portal PR soon.

It replicates the old functionality, but also:
  - Updates automatically (can be disabled).
  - Works better on small screens. The old page was trying to do it, but it looked pretty much broken.

It's a bit different than other components in this repo. Some thoughts about architecture:

1. Webpack is now configured to handle ES6, JSX, CSS modules and assets loading, so I've made use of that. I think it makes sense to create a separate directory for every new page / component groups and keep JS together with CSS and related assets. Things evolve pretty quickly, it will be easier to have those mini-apps in separate directories rather than keep updating repo structure all the time. Plus, it's easier to track dead code and find related assets and styles. And it's not an unusual pattern, even `create-react-app` uses similar structure (JS and CSS together).

2. The top-level component doesn't have any UI specific code (`index.js`). All it does is to handle and process data and pass it down to "dumb" components. That should the only state handling component per app / page if possible (except some tiny UI states like modal visibility, hide/expand some section, and so on). I think it's a good pattern for apps where redux / mobx would be an overkill, it worked for me pretty well even for way bigger apps.

3. Top-level component should accept `initialData` as a property. In that case, it shouldn't request any data from Portal on the initial load. It's easy to do and will ensure that it's easy to test a component using mock data + provide pseudo server-side rendering if necessary to cut additional AJAX requests.

4. All the global Portal references should be in `defaultProps`. Makes easier to test in the future. That's already described in PortalPages docs. Actually, I'd say that we could even say that there should be no dependencies on Portal code and everything should be set up on Portal side. It makes it a bit more correct IMHO, although `defaultProps` make the component a bit easier to reuse and still provide a way to make it testable (if we overwrite all of them).

5. CSS modules make nesting unnecessary, but I think it's still helpful. We could either keep styles in multiple files (one per component) or one per directory. I used the latter, as there's not much CSS here. But I still keep components separate as a top-level namespace, so it's easy to divide if necessary.

6. I've added a test page with mock data. We can display it using dev-server. I'd say it's pretty confusing (you need to pick some random portal URL and selector, it can be even `body`), but I'd rather work on testing separately.

If it makes sense to you, we could add some of that to the Portal Pages guide doc.

### CODE REVIEW

**Comment by pjanik** [comment]
Some screenshots with a comparison, the new version is on the right:

1. screen shot 2018-04-23 at 14 49 50
2. screen shot 2018-04-23 at 14 57 03
3. screen shot 2018-04-23 at 14 57 53

File: src/library/components/recent-activity/offering.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+            </div>
+            <div className={css.inProgress} style={{width: `${inProgressWidth}%`}} >
+              { inProgressStudentsCount > 0 && <div className={css.count}>{ inProgressStudentsCount }</div> }
+            </div>
+            <div className={css.notStarted} style={{width: `${notStartedWidth}%`}} >
+              { notStartedStudentsCount > 0 && <div className={css.count}>{ notStartedStudentsCount }</div> }
+            </div>
+          </div>
+        </div>
+        <a href={reportPath} target='_blank' className={css.reportBtn + ' button'}>Report</a>
```
[the remark this answers is not in the document]
Comment (scytacki) [reply src/library/components/recent-activity/offering.js]: These reports can be at the offering, class, or teacher level. And they can specify what the label of the extra report button should be.

**Comment by pjanik** [comment]
> I might be missing it, but it looks like this code doesn't have the 'alternative report' or 'external report' option.

That's right, thanks. The last commit fixes that.

**Comment by scytacki** [comment]
Looks great.  Yes please add this to the Portal Pages guide doc.

**Comment by pjanik** [comment]
I've updated rules in Portal Pages guide: https://docs.google.com/document/d/167jhX9vRrNvGCI5QW7dfw3uJ38Jc7NpDGlpFAXziYlI/edit#

**Pull request merged**